### PR TITLE
doc: document NMT wrapper in prepare_proposal

### PR DESCRIPTION
**Pull request by SweeXordious**
Adds a small note to specify that the NMT wrapper is used to extend the shares

### CODE REVIEW

File: pkg/da/data_availability_header.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ import (
 	"fmt"
 
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
-	"github.com/celestiaorg/celestia-app/pkg/wrapper"
+	nmtwrapper "github.com/celestiaorg/celestia-app/pkg/wrapper"
```
Comment (evan-forbes) [inline pkg/da/data_availability_header.go:9]: [not blocking comment]
I like this change, but we use wrapper everywhere else. Not going to block on this, but perhaps in the future we should rename the package to force this across all repos

File: pkg/da/data_availability_header.go [hunk 2]
```diff
@@ -68,7 +68,8 @@ func ExtendShares(squareSize uint64, shares [][]byte) (*rsmt2d.ExtendedDataSquar
 		)
 	}
 	// here we construct a tree
-	return rsmt2d.ComputeExtendedDataSquare(shares, appconsts.DefaultCodec(), wrapper.NewConstructor(squareSize))
+	// Note: uses the nmt wrapper to construct the tree.
+	return rsmt2d.ComputeExtendedDataSquare(shares, appconsts.DefaultCodec(), nmtwrapper.NewConstructor(squareSize))
```
Comment (evan-forbes) [inline pkg/da/data_availability_header.go:72]: [nit non-blocking]
perhaps we could merge this we the above outdated comment (that doesn't really make sense by itself)?

`// we construct the data root using the nmt wrapper`

File: app/prepare_proposal.go [hunk 3]
```diff
@@ -59,6 +59,8 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 	}
 
 	// erasure the data square which we use to create the data root.
+	// Note: uses the nmt wrapper to construct the tree.
+	// checkout pkg/wrapper/nmt_wrapper.go for more information.
```
Comment (evan-forbes) [inline app/prepare_proposal.go:63]: [I'll do this in a different PR]
oh woah we should definitely add docs to https://github.com/celestiaorg/celestia-app/blob/24626ff8d00854d7e0027abcfdf50de1a2aee6a3/pkg/da/data_availability_header.go#L52

**Review by evan-forbes — APPROVED** [review]

File: pkg/da/data_availability_header.go (hunk 2 above)
Comment (SweeXordious) [reply pkg/da/data_availability_header.go:72]: `// Note: uses the nmt wrapper to construct the tree.` => `// we construct the data root using the nmt wrapper`?

File: pkg/da/data_availability_header.go (hunk 1 above)
Comment (SweeXordious) [reply pkg/da/data_availability_header.go:9]: I'll create an issue for this: https://github.com/celestiaorg/celestia-app/issues/1069

File: app/prepare_proposal.go (hunk 3 above)
Comment (SweeXordious) [reply app/prepare_proposal.go:63]: Thanks :+1:

**Comment by evan-forbes** [comment]
sure why not, but lets stay consistent with the naming https://github.com/celestiaorg/celestia-app/issues/1069#issuecomment-1327702847

File: pkg/da/data_availability_header.go (hunk 2 above)
Comment (evan-forbes) [reply pkg/da/data_availability_header.go:73]: yep, makes sense!

File: pkg/da/data_availability_header.go (hunk 1 above)
Comment (evan-forbes) [reply pkg/da/data_availability_header.go:10]: since #1069 is unplanned, do you mind reverting this?

**Review by evan-forbes — COMMENTED** [review]
apologies for letting this slip through the cracks @SweeXordious 

left one comment to use the same name for the wrapper across the board, then should finally merge

**Comment by SweeXordious** [comment]
done, all we need is an approve :D

File: pkg/da/data_availability_header.go (hunk 1 above)
Comment (SweeXordious) [reply pkg/da/data_availability_header.go]: done

**Pull request merged**